scripts/lsp-generate-bindings.el

```diff
@@ -0,0 +1,99 @@
+  (insert-file-contents-literally "generated.protocol.schema.json")
+  (->> (plist-get (json-parse-string (buffer-string)
+                                     :object-type 'plist
+                                     :null-object nil
+                                     :false-object nil)
+                  :$defs)
+
+       (-partition 2)
+       (-filter (-lambda ((_ (&plist :type)))
+                  (string= type "object")))
+       (-map (-lambda ((name type-data))
+               (let* ((key (intern (substring (symbol-name name) 1)))
+                      (required (-map (lambda (field-name)
+                                        (intern (concat ":" field-name))) (plist-get type-data :required)))
+                      (params (->> (plist-get type-data :properties)
+                                   (-partition 2)
+                                   (-map #'cl-first)
+                                   (-filter (-not (-partial #'-contains? required))))))
+                 (list key required params))))
+       (apply #'nconc '(lsp-interface ))))
+
+(with-temp-buffer
+  (insert-file-contents-literally "generated.protocol.schema.json")
+
+  (let ((defs (plist-get (json-parse-string (buffer-string)
+                                            :object-type 'plist
+                                            :null-object nil
+                                            :false-object nil)
+                         :$defs)))
+    `(progn
+       ,@(->> defs
+              (-partition 2)
+              (-filter (-lambda ((_ (&plist :type)))
+                         (not (string= type "object"))))
+
+              (-map (-lambda ((type (&plist :enum)))
+                      (seq-map-indexed (-lambda (value index)
+                                         (list 'defconst (intern (format "lsp/%s-%s"
+                                                                         (s-dashed-words (substring (symbol-name type) 1))
+                                                                         (s-dashed-words  value)))
+                                               index))
+                                       enum)))
+              (apply #'nconc))
+
+       ,(->> defs
+             (-partition 2)
+             (-filter (-lambda ((_ (&plist :type)))
+                        (string= type "object")))
+             (-map (-lambda ((name type-data))
+                     (let* ((key (intern (substring (symbol-name name) 1)))
+                            (required (-map (lambda (field-name)
+                                              (intern (concat ":" field-name))) (plist-get type-data :required)))
+                            (params (->> (plist-get type-data :properties)
+                                         (-partition 2)
+                                         (-map #'cl-first)
+                                         (-filter (-not (-partial #'-contains? required))))))
+                       (list key required params))))
```

Comment:
I think with this `list`, the return value will be in the form of `(lsp-interface (key required params) ....)`? Not flattened list like in `lsp-protocol`?

Reply:
The next line will merge the lists